Mathlib/Algebra/Order/BigOperators/Expect.lean

```diff
@@ -0,0 +1,186 @@
+    𝔼 i ∈ s, f i = 0 ↔ ∀ i ∈ s, f i = 0 := by
+  simp [expect, sum_eq_zero_iff_of_nonpos hf, hs.ne_empty]
+
+section PosSMulMono
+variable [PosSMulMono ℚ≥0 α] {a : α}
+
+lemma expect_le_expect (hfg : ∀ i ∈ s, f i ≤ g i) : 𝔼 i ∈ s, f i ≤ 𝔼 i ∈ s, g i :=
+  smul_le_smul_of_nonneg_left (sum_le_sum hfg) <| by positivity
+
+/-- This is a (beta-reduced) version of the standard lemma `Finset.expect_le_expect`,
+convenient for the `gcongr` tactic. -/
+@[gcongr]
+lemma _root_.GCongr.expect_le_expect (h : ∀ i ∈ s, f i ≤ g i) : s.expect f ≤ s.expect g :=
+  Finset.expect_le_expect h
+
+lemma expect_le (hs : s.Nonempty) (h : ∀ x ∈ s, f x ≤ a) : 𝔼 i ∈ s, f i ≤ a :=
+  (inv_smul_le_iff_of_pos <| mod_cast hs.card_pos).2 <| by
+    rw [Nat.cast_smul_eq_nsmul]; exact sum_le_card_nsmul _ _ _ h
+
+lemma le_expect (hs : s.Nonempty) (h : ∀ x ∈ s, a ≤ f x) : a ≤ 𝔼 i ∈ s, f i :=
+  (le_inv_smul_iff_of_pos <| mod_cast hs.card_pos).2 <| by
+    rw [Nat.cast_smul_eq_nsmul]; exact card_nsmul_le_sum _ _ _ h
+
+lemma expect_nonneg (hf : ∀ i ∈ s, 0 ≤ f i) : 0 ≤ 𝔼 i ∈ s, f i :=
+  smul_nonneg (by positivity) <| sum_nonneg hf
+
+end PosSMulMono
+
+section PosSMulMono
+variable {M N : Type*} [AddCommMonoid M] [Module ℚ≥0 M] [OrderedAddCommMonoid N] [Module ℚ≥0 N]
+  [PosSMulMono ℚ≥0 N]
+
+-- TODO: Contribute back better docstring to `le_prod_of_submultiplicative`
+/-- If `m` is a subadditive function (`m (x + y) ≤ m x + m y`, `f 1 = 1`), and `f i`,
+`i ∈ s`, is a finite family of elements, then `m (𝔼 i in s, f i) ≤ 𝔼 i in s, m (f i)`. -/
+lemma le_expect_of_subadditive (m : M → N) (h_zero : m 0 = 0)
+    (h_add : ∀ a b, m (a + b) ≤ m a + m b) (h_div : ∀ a (n : ℕ), m (a /ℚ n) = m a /ℚ n)
+    (s : Finset ι) (f : ι → M) : m (𝔼 i ∈ s, f i) ≤ 𝔼 i ∈ s, m (f i) := by
+  simp only [expect, h_div]
+  exact smul_le_smul_of_nonneg_left (le_sum_of_subadditive _ h_zero h_add _ _) <| by positivity
+
+end PosSMulMono
+end OrderedAddCommMonoid
+
+section OrderedCancelAddCommMonoid
+variable [OrderedCancelAddCommMonoid α] [Module ℚ≥0 α] {s : Finset ι} {f g : ι → α}
+section PosSMulStrictMono
+variable [PosSMulStrictMono ℚ≥0 α]
+
+lemma expect_pos (hf : ∀ i ∈ s, 0 < f i) (hs : s.Nonempty) : 0 < 𝔼 i ∈ s, f i :=
+  smul_pos (inv_pos.2 <| mod_cast hs.card_pos) <| sum_pos hf hs
+
+end PosSMulStrictMono
+end OrderedCancelAddCommMonoid
+
+section LinearOrderedAddCommMonoid
+variable [LinearOrderedAddCommMonoid α] [Module ℚ≥0 α] [PosSMulMono ℚ≥0 α] {s : Finset ι}
+  {f : ι → α} {a : α}
+
+lemma lt_of_lt_expect (hs : s.Nonempty) (h : a < 𝔼 i ∈ s, f i) : ∃ x ∈ s, a < f x := by
+  contrapose! h; exact expect_le hs h
+
+lemma lt_of_expect_lt (hs : s.Nonempty) (h : 𝔼 i ∈ s, f i < a) : ∃ x ∈ s, f x < a := by
+  contrapose! h; exact le_expect hs h
+
+end LinearOrderedAddCommMonoid
+
+section LinearOrderedAddCommGroup
+variable [LinearOrderedAddCommGroup α] [Module ℚ≥0 α] [PosSMulMono ℚ≥0 α]
+
+lemma abs_expect_le_expect_abs (s : Finset ι) (f : ι → α) : |𝔼 i ∈ s, f i| ≤ 𝔼 i ∈ s, |f i| :=
```

Comment:
norm version? also can be todo

Reply:
It's in APAP already. Can add it, but it will become a new file